From 61e70e115f13f578294cd6dba6a3faaeaf9c0867 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Tue, 18 Mar 2014 09:18:47 -0700
Subject: [PATCH 491/827] media: xilinx: tpg: Use xvip_enum_mbus_code()

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Use xvip_enum_mbus_code() instead of own function, xtpg_enum_mbus_code().

Signed-off-by: Hyun Kwon <hyunk@xilinx.com>
Acked-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit c38eb2fa500a7cf94baab215a5185c359e063b83)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-tpg.c | 16 +---------------
 1 file changed, 1 insertion(+), 15 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-tpg.c b/drivers/media/platform/xilinx/xilinx-tpg.c
index fda5cfd..1dfc281 100644
--- a/drivers/media/platform/xilinx/xilinx-tpg.c
+++ b/drivers/media/platform/xilinx/xilinx-tpg.c
@@ -159,20 +159,6 @@ static int xtpg_s_stream(struct v4l2_subdev *subdev, int enable)
  * V4L2 Subdevice Pad Operations
  */
 
-static int xtpg_enum_mbus_code(struct v4l2_subdev *subdev,
-			       struct v4l2_subdev_fh *fh,
-			       struct v4l2_subdev_mbus_code_enum *code)
-{
-	struct xtpg_device *xtpg = to_tpg(subdev);
-
-	if (code->index)
-		return -EINVAL;
-
-	code->code = xtpg->vip_format->code;
-
-	return 0;
-}
-
 static int xtpg_enum_frame_size(struct v4l2_subdev *subdev,
 				struct v4l2_subdev_fh *fh,
 				struct v4l2_subdev_frame_size_enum *fse)
@@ -349,7 +335,7 @@ static struct v4l2_subdev_video_ops xtpg_video_ops = {
 };
 
 static struct v4l2_subdev_pad_ops xtpg_pad_ops = {
-	.enum_mbus_code		= xtpg_enum_mbus_code,
+	.enum_mbus_code		= xvip_enum_mbus_code,
 	.enum_frame_size	= xtpg_enum_frame_size,
 	.get_fmt		= xtpg_get_format,
 	.set_fmt		= xtpg_set_format,
-- 
2.9.3

